### PR TITLE
Users/suaggar/warcustomname

### PR DESCRIPTION
**Pull request by SumiranAgg**

### CODE REVIEW

File: Tasks/AzureWebAppV1/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -18,6 +18,8 @@
   "loc.input.help.slotName": "Enter or Select an existing Slot other than the Production slot.",
   "loc.input.label.package": "Package or folder",
   "loc.input.help.package": "File path to the package or a folder containing app service contents generated by MSBuild or a compressed zip or war file.<br />Variables ( [Build](https://docs.microsoft.com/vsts/pipelines/build/variables) | [Release](https://docs.microsoft.com/vsts/pipelines/release/variables#default-variables)), wildcards are supported. <br/> For example, $(System.DefaultWorkingDirectory)/\\*\\*/\\*.zip or $(System.DefaultWorkingDirectory)/\\*\\*/\\*.war.",
+  "loc.input.label.customWarName": "Custom war name",
+  "loc.input.help.customWarName": "Specify custom folder name you want to deploy to. <br/>If the field is empty, package is deployed to <appname>.azurewebsites.net/<warpackagename><br/> If ROOT is entered, package is deployed to <appname>.azurewebsited.net <br/>else it is deployed to <appname>.azurewebsited.net/<customWarName>",
```
Comment (vincentdass) [inline Tasks/AzureWebAppV1/Strings/resources.resjson/en-US/resources.resjson]: Can you please get the sign off for resources from @N-Usha?

File: Tasks/AzureFunctionAppV1/Strings/resources.resjson/en-US/resources.resjson [hunk 2]
```diff
@@ -18,6 +18,8 @@
   "loc.input.help.slotName": "Enter or Select an existing Slot other than the Production slot.",
   "loc.input.label.package": "Package or folder",
   "loc.input.help.package": "File path to the package or a folder containing app service contents generated by MSBuild or a compressed zip or war file.<br />Variables ( [Build](https://docs.microsoft.com/vsts/pipelines/build/variables) | [Release](https://docs.microsoft.com/vsts/pipelines/release/variables#default-variables)), wildcards are supported. <br/> For example, $(System.DefaultWorkingDirectory)/\\*\\*/\\*.zip or $(System.DefaultWorkingDirectory)/\\*\\*/\\*.war.",
+  "loc.input.label.customWarName": "Custom war name",
```
Comment (vincentdass) [inline Tasks/AzureFunctionAppV1/Strings/resources.resjson/en-US/resources.resjson]: I don't think function app takes WAR package. Can you please verify it once?

File: Tasks/AzureWebAppV1/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts [hunk 3]
```diff
@@ -45,7 +45,7 @@ export class BuiltInLinuxWebAppDeploymentProvider extends AzureRmWebAppDeploymen
 
             case PackageType.war:
                 tl.debug("Initiated deployment via kudu service for webapp war package : "+ this.taskParams.Package.getPath());
-                let warName = webCommonUtility.getFileNameFromPath(this.taskParams.Package.getPath(), ".war");
+                let warName = this.taskParams.CustomWarName || webCommonUtility.getFileNameFromPath(this.taskParams.Package.getPath(), ".war");
```
Comment (vincentdass) [inline Tasks/AzureWebAppV1/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts]: can we print this value in debug logs?

File: Tasks/AzureWebAppV1/Strings/resources.resjson/en-US/resources.resjson (hunk 1 above)
Comment (SumiranAgg) [reply Tasks/AzureWebAppV1/Strings/resources.resjson/en-US/resources.resjson]: done

**Pull request merged**